source/slang/slang-lookup.cpp

```diff
@@ -449,6 +449,12 @@ static void _lookUpMembersInSuperTypeDeclImpl(
             if (int(request.options) & int(LookupOptions::IgnoreBaseInterfaces))
                 continue;
         }
+        // If we are looking up only immediate members, ignore non "Self" facets
+        else if (int(request.options) & int(LookupOptions::IgnoreInheritance) 
```

Comment:
What about extensions?

struct A : B
{ int member1; }

extension A { int member2; }

Do we want to consider member2?

Reply:
I will add a test for this and implement accordingly

Reply:
I was looking into how to implement the following, I believe the least "hacked" code would be to have `IgnoreInheritance`, and use lookup to find a default constructor.